Test bugs surfaced after #423 fix: tests rely on cwd pollution

## Summary

After #423 fixed cross-test pollution by restoring `os.getcwd()` between tests, two tests that previously passed (only by accident — they depended on a previous test having `chdir`-ed into a problem package) now fail in isolation. They never had their own `cleandir`/`pkg_cleandir` fixture and call into `package.find_problem()` which requires cwd to be inside a problem.

(Filing separately to keep #423 focused. The `tests/rbx/box/packaging/e2e/test_boca_e2e.py` failures are excluded — those are tracked elsewhere.)

## Failing tests

### 1. `tests/rbx/box/testcase_utils_test.py::TestClearBuiltTestcases::test_clear_built_testcases_nonexistent`

```
def test_clear_built_testcases_nonexistent(self):
    """Test clearing when directory doesn't exist doesn't raise error."""
    # This should not raise an error
    clear_built_testcases()
```

Traceback:
```
rbx/box/testcase_utils.py:95: in clear_built_testcases
    shutil.rmtree(str(get_build_tests_path()), ignore_errors=True)
rbx/box/package.py:260: in get_build_path
    return find_problem(root) / environment.get_build_dir()
rbx/box/package.py:87: in find_problem
    raise typer.Exit(1)
E   click.exceptions.Exit: 1
Captured stdout: Problem not found in /Users/rsalesc/Dev/robox.io
```

The test docstring says it should *not* raise. Two possible fixes:
- Make `clear_built_testcases()` tolerate \"no problem found\" (probably the right call given the docstring intent).
- Or add `pkg_cleandir` / `pkg_from_testdata` fixture so the test runs inside a problem.

### 2. `tests/rbx/box/test_testcase_sample_utils.py::test_get_statement_samples_interaction_generated`

```python
@pytest.mark.asyncio
async def test_get_statement_samples_interaction_generated(
    tmp_path, mock_extract_generation_testcases_from_groups
):
    \"\"\"Test 14: Interaction (Generated).\"\"\"
```

Same failure mode — calls `package.get_statement_chunks_folder()` which traverses to `find_problem()`. Test takes only `tmp_path` and a mock fixture; doesn't `chdir` into a package. Fix: switch to `pkg_cleandir` / `pkg_from_testdata` like its siblings, or have it set up its own problem dir.

## Reproduction

```bash
uv run pytest \
  tests/rbx/box/testcase_utils_test.py::TestClearBuiltTestcases::test_clear_built_testcases_nonexistent \
  tests/rbx/box/test_testcase_sample_utils.py::test_get_statement_samples_interaction_generated
# → 2 failed
```

Also fails under any random ordering after #423 lands.

## Why this is the right kind of failure to expose

These were latent bugs masked by #423's pollution. The fix is to make the tests honest about their fixtures, not to bring back the pollution.

--- a/tests/rbx/box/test_testcase_sample_utils.py
+++ b/tests/rbx/box/test_testcase_sample_utils.py
@@ -8,6 +8,7 @@ from rbx.box import testcase_sample_utils
 from rbx.box.generation_schema import GenerationMetadata, GenerationTestcaseEntry
 from rbx.box.schema import Testcase
 from rbx.box.testcase_schema import TestcaseEntry
+from rbx.box.testing import testing_package
 
 
 @pytest.fixture
@@ -323,7 +324,9 @@ async def test_get_statement_samples_out_statement_vs_out(
 
 @pytest.mark.asyncio
 async def test_get_statement_samples_interaction_generated(
-    tmp_path, mock_extract_generation_testcases_from_groups
+    tmp_path,
+    mock_extract_generation_testcases_from_groups,
+    testing_pkg: testing_package.TestingPackage,
 ):
     """Test 14: Interaction (Generated)."""
     dest_input = tmp_path / 'dest.in'

--- a/tests/rbx/box/testcase_utils_test.py
+++ b/tests/rbx/box/testcase_utils_test.py
@@ -217,7 +217,9 @@ class TestClearBuiltTestcases:
 
         assert not tests_path.exists()
 
-    def test_clear_built_testcases_nonexistent(self):
+    def test_clear_built_testcases_nonexistent(
+        self, testing_pkg: testing_package.TestingPackage
+    ):
         """Test clearing when directory doesn't exist doesn't raise error."""
         # This should not raise an error
         clear_built_testcases()

